### PR TITLE
Feature 12 Show map on Library Info

### PR DESCRIPTION
**Pull request by lucas-mrq**

### CODE REVIEW

File: app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java [hunk 1]
```diff
@@ -101,6 +144,17 @@ public void onFetchFailed() {
...
         });
 
+        Button routeButton = findViewById(R.id.routeButton);
+        routeButton.setOnClickListener(view -> {
+            startGoogleMap();
+        });
+
+        Button centerButton = findViewById(R.id.centerButton);
+        centerButton.setOnClickListener(view -> {
+            int padding = 100;
```
Comment (Vio1625) [inline app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java]: Change the padding here

File: app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java [hunk 2]
```diff
@@ -161,4 +234,76 @@ public void onFailure(Call<List<Book>> call, Throwable t) {
...
+                    List<Library> libraries = response.body();
+                    for (Library library : libraries) {
+                        if (library.id == libraryId) {
+                            libraryLocation = new LatLng(library.latitude, library.longitude);
+                            mGoogleMap.addMarker(new MarkerOptions().position(libraryLocation).title(library.name));
+                            LatLngBounds.Builder builder = new LatLngBounds.Builder();
+                            builder.include(libraryLocation);
+                            bounds = builder.build();
+                            int padding = 100; // Adjust the padding as needed
+                            mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngBounds(bounds, padding));
```
Comment (Vio1625) [inline app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java]: Padding of 15 would be probably better
mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latlng, 15));

File: app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java (hunk 1 above)
Comment (lucas-mrq) [reply app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java]: Done

File: app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java (hunk 2 above)
Comment (lucas-mrq) [reply app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java]: Done

**Review by Vio1625 — APPROVED** [review]
Good job !

**Pull request merged**